Pocket lab updates

mphalke opened the pull request:
pocket_lab: Bumped up stack size for string variables
  Increased the stack size of variables dealing with attribute read/write
  to 100 bytes from 50 bytes to avoid potential issue with less memory
  to store attribute related strings

  pocket_lab: Fixed issue with attribute reading
  Device attribute getter/setter APIs have extra argument of channel info
  which is not applicable for device specific attributes but has been
  utilized in IIO app layer/firmware for having common API implementation
  for both channel and device attribute getter/setters. This is why
  channel info local structure variable is explicitly initialized with 0

  pocket_lab: Fixed issue with DMM measurements
  Added missing integer typecasting for intger and non-integer
  numbers arithmetic

  pocket_lab: Fixed the issue with handling FFT samples size
  1. Number of FFT samples are processed from FFT init parameter structures supplied
  through application layer. Earlier default sample size was used for processing the data
  2. Changed fft processing and measurement structures from pointer to variables
  so that memory is allocated from RAM rather than heap. Heap allocation results
  into run time failures as FFT structures are huge in size

  fft: Removed heap (dynamic memory) dependency in fft library
  1. Removed the heap allocation for FFT parameters structures from library.
  The fft parameter structues are of length equal to number of FFT samples.
  Allocating such a huge memory from heap results into run time failures
  when heap memory run out of space. Memory to these structures is allocated
  from application layer, which could be either heap or just RAM.
  2. Reduced the max default FFT samples size to 2048 as few existing FW
  applications already consumes huge amount of memory and so no sufficient
  memory is left for FFT parameters storage. Also, the condition check has been
  added for the macro so that FFT samples size selection can be done from
  build system

CMinajigi commented on pocket_lab/pl_gui_iio_wrapper.c:
```diff
@@ -274,7 +274,7 @@ int32_t pl_gui_get_chn_attr_avail_options(const char *attr_name,
 {
 	struct iio_device *iio_dev;
-	struct iio_ch_info chn_info;
+	struct iio_ch_info chn_info = { 0 };
```
Is channel info is not applicable here as it is get channel attribute method?

mphalke replied:
I think I missed to update chn_info later. I just need channel number here. Will update this

mphalke commented:
Hi All, any comments?

CMinajigi commented on pocket_lab/pl_gui_iio_wrapper.c:
```diff
@@ -363,7 +363,7 @@ int32_t pl_gui_read_chn_attr(char *attr_name, char *attr_val,
 {
 	struct iio_device *iio_dev;
-	struct iio_ch_info chn_info;
+	struct iio_ch_info chn_info = { 0 };
```
Is channel info is not applicable here also?

mphalke replied:
chn_info.ch_num = chn_indx;
This sets the channel number later in the function. I only need channel number for attributes read/write

mphalke commented:
V1.0: Added missing initialization of chn_info parameter
-Added missing initialization of channel info with channel number (index) in pl_gui_get_chn_attr_avail_options() function

ribdp commented on pocket_lab/pl_gui_views.h:
```diff
@@ -108,9 +108,10 @@ int32_t pl_gui_create_dmm_view(lv_obj_t* parent,
-			       struct pl_gui_init_param *param);
+				    struct pl_gui_init_param *param);
 int32_t pl_gui_create_about_view(lv_obj_t* parent,
 				 struct pl_gui_init_param *param);
```
nit: should this be right indented?

mphalke replied:
This is what I get when I apply astyle even though I manually try to intend to right. I would keep what astyle does

ribdp commented on pocket_lab/pl_gui_events.c:
```diff
@@ -124,10 +124,10 @@ int32_t pl_gui_event_read(uint8_t *buf, uint32_t len)
 				}
 
 				/* Get total bytes based on the fixed samples count */
```
nit: is this comment still valid?

mphalke replied:
no, will remove this

mphalke commented:
V2.0: Removed redundant comment in pl_gui_events.c file
